kf_lib_data_ingest/target_api_plugins/kids_first_dataservice.py

```diff
@@ -0,0 +1,747 @@
+            "study_id": record[CONCEPT.STUDY.ID],
+            "family_id": get_target_id_from_record(Family, record),
+            "external_id": key,
+            "is_proband": record.get(CONCEPT.PARTICIPANT.IS_PROBAND),
+            "ethnicity": record.get(CONCEPT.PARTICIPANT.ETHNICITY),
+            "gender": record.get(CONCEPT.PARTICIPANT.GENDER),
+            "race": record.get(CONCEPT.PARTICIPANT.RACE),
+            "affected_status": record.get(
+                CONCEPT.PARTICIPANT.IS_AFFECTED_UNDER_STUDY
+            ),
+            "species": record.get(CONCEPT.PARTICIPANT.SPECIES),
+            "visible": record.get(CONCEPT.PARTICIPANT.VISIBLE),
+        }
+
+
+class Diagnosis:
+    class_name = "diagnosis"
+    api_path = "/diagnoses"
+    target_id_concept = CONCEPT.DIAGNOSIS.TARGET_SERVICE_ID
+    id_fields = {"kf_id", "participant_id"}
+
+    @staticmethod
+    def build_key(record):
+        assert None is not record[CONCEPT.PARTICIPANT.ID]
+        assert None is not record[CONCEPT.DIAGNOSIS.NAME]
+        return record.get(CONCEPT.DIAGNOSIS.UNIQUE_KEY) or join(
+            record[CONCEPT.PARTICIPANT.ID],
+            record[CONCEPT.DIAGNOSIS.NAME],
+            record.get(CONCEPT.DIAGNOSIS.EVENT_AGE_DAYS),
+        )
+
+    @staticmethod
+    def build_entity(record, key, get_target_id_from_record):
+        return {
+            "kf_id": get_target_id_from_record(Diagnosis, record),
+            "participant_id": get_target_id_from_record(Participant, record),
+            "external_id": key,
+            "age_at_event_days": record.get(CONCEPT.DIAGNOSIS.EVENT_AGE_DAYS),
+            "source_text_diagnosis": record.get(CONCEPT.DIAGNOSIS.NAME),
+            "source_text_tumor_location": record.get(
+                CONCEPT.DIAGNOSIS.TUMOR_LOCATION
+            ),
+            "mondo_id_diagnosis": record.get(CONCEPT.DIAGNOSIS.MONDO_ID),
+            "icd_id_diagnosis": record.get(CONCEPT.DIAGNOSIS.ICD_ID),
+            "uberon_id_tumor_location": record.get(
+                CONCEPT.DIAGNOSIS.UBERON_TUMOR_LOCATION_ID
+            ),
+            "ncit_id_diagnosis": record.get(CONCEPT.DIAGNOSIS.NCIT_ID),
+            "spatial_descriptor": record.get(
+                CONCEPT.DIAGNOSIS.SPATIAL_DESCRIPTOR
+            ),
+            "diagnosis_category": record.get(CONCEPT.DIAGNOSIS.CATEGORY),
+            "visible": record.get(CONCEPT.DIAGNOSIS.VISIBLE),
+        }
+
+
+class Phenotype:
+    class_name = "phenotype"
+    api_path = "/phenotypes"
+    target_id_concept = CONCEPT.PHENOTYPE.TARGET_SERVICE_ID
+    id_fields = {"kf_id", "participant_id"}
+
+    @staticmethod
+    def build_key(record):
+        assert None is not record[CONCEPT.PARTICIPANT.ID]
+        assert None is not record[CONCEPT.PHENOTYPE.NAME]
+        assert record[CONCEPT.PHENOTYPE.OBSERVED] in {
+            constants.PHENOTYPE.OBSERVED.NO,
+            constants.PHENOTYPE.OBSERVED.YES,
```

Comment:
The previous target api requires [`CONCEPT.PHENOTYPE.OBSERVED`](https://github.com/kids-first/kf-lib-data-ingest/blob/master/kf_lib_data_ingest/target_apis/kids_first.py#L481), not confining only to `'Positive'` or `'Negative'`. We may just do `assert None is not record[CONCEPT.PHENOTYPE.OBSERVED]`?

Reply:
True, but I think we shouldn't actually load the phenotypes otherwise.